### ISSUE TITLE
The program takes a long time to open, and my system keeps freezing and memory consumption is very high.

### ISSUE THREAD

**Issue by martinsjadie** [X-Hax/sadx-mod-loader#150]
Hello everything is fine?

I come through this statement, and try to explain the difficulty I am going through with the program.

Today I installed the BetterSADX mod pack, after completing the program installation. Opened the old version and asked to install a new version of the mod manager, due to the current HD-GUI mod version 0.964 no longer supporting the old SA Mod Manager.

The bug is that it's difficult to open the program, my RAM consumption increases to the maximum. And when I close the program, the process continues running through the task manager and the RAM is full of my 4GB that I have on my laptop.

And when I try to finalize the program's process tree, it gives an access denied error.

But after a few minutes the process closes.

This situation happened the first time with the PkR Mod Installer, with the mod manager. I thought it was with the mods but I don't think that's the case.

Due to my laptop being old, I chose to download SADX via the PC version and not have to play via the Dreamcast emulator.

My Laptop Specifications:

LG C400

Processor: Intel Pentium P6200 Dual Core 2.13Ghz
Motherboard: LG C400-G.BC23P1
Ram memory: 2x2gb DDR3 with a frequency of 1333Mhz.
Hard Drive: TOSHIBA MK5059GSXP 500Gb Sata 2.5
Operating System: Windows 7 Professional x64 bits.

**Comment by martinsjadiel21** [comment]
Yes my friend, in the old version of Mod Loader, there were no performance problems when opening and loading the program.

I also tested it on the 2004 Vanilla version.

More or less how it happens, I'll explain it again.

In the SADX 2004 version, which I had and uninstalled out of annoyance and sadness, I did the same thing. I updated the Mod Loader to the current version and when opening it, my Windows 7 system freezes and the mouse pointer is so slow that I can't even follow the task manager or try to open other tasks.

After 1~3 minutes, the program opens. Only then can I touch it, configure something and be able to play the game.

Incredible as it may seem, memory usage stabilizes normally.

Then when I close the program the RAM consumption is 3.58GB, so the process is difficult to close. And when I try to finalize the process tree, access is denied. After a few minutes, it shuts down on its own and RAM consumption stabilizes normally.

I'm going to test with the 2004 version only with the old Mod Loader, to see if it will give an error or not.

I even thought about recording a video or taking photos to show the problem. If it does not violate the platform's rules.

**Comment by martinsjadiel21** [comment]
![Sem título](https://github.com/user-attachments/assets/8f0d2d12-3767-43c0-8be5-62a2bf976cb3)
![Sem título1](https://github.com/user-attachments/assets/9fd0a59b-56b7-4d48-b1d0-dee6655518c0)

I managed to solve the problem with ModLoader.

With the Mod Installer pack, I went to the "instdata" folder, went to the "SAModManager_x86.zip" file and opened it. I got the executable and put it in the game's root folder. And it opened right away and I had no crashes, no high RAM consumption or anything.

But even so, when I go to open the modloader, a new update always appears. And when I update the program to the new version, the problem comes back.

What I'm worried about is that my processor is x64 bits, and this wasn't supposed to happen.

In this case, the ModLoader version is 1.2.10 (f185c67), as shown in the prints. And I run it without problems, and I just cancel the updates. Because it already runs without problems.

I think this is my case, I don't know if this crazy situation has happened to another user.

**Comment by PiKeyAr** [comment]
1.2.10 is the previous release of the Mod Manager from June (you seem to have an outdated archive of the Mod Installer, since it refreshes the Mod Manager automatically every week).

Can you try with these? If none of them work, it could be some issue introduced recently.
https://dcmods.unreliable.network/owncloud/data/PiKeyAr/files/Setup/data/SAModManager_x86.zip
https://dcmods.unreliable.network/owncloud/data/PiKeyAr/files/Setup/data/SAModManager_x64.zip

**Comment by martinsjadiel21** [comment]
It opened right away and without freezing my system, it was version:
x86 - Version (1.2.11 - cc3ac2c).

As for x64, the problem appeared right away.
